packages/hardhat-core/test/internal/hardhat-network/provider/modules/hardhat.ts

```diff
@@ -1382,7 +1387,8 @@ describe("Hardhat module", function () {
           assert.lengthOf(pendingTxsAfter, 0);
         });
 
-        describe("tests using sinon", () => {
+        // TODO: https://github.com/NomicFoundation/edr/issues/249
+        describe.skip("tests using sinon", () => {
```

Comment:
Is there a blocker to make this work?

Reply:
Yes, I discussed this in private with @fvictorio. In JS/TS there is a way to override the system clock and manually control it using `sinon`. This doesn't however affect the actual system clock, which means that EDR still uses the actual time.

The only way to support similar functionality would be to allow users to pass in a type that implements a custom `trait Clock` which allows the end user to have control over the clock used by EDR. By default, this would use an implementation with `SystemClock`

Franco recommended to ignore these tests for now and revisit the issue later

Reply:
Agreed with ignoring these for now.

We should find a way to validate it before the release though. Maybe port these tests to rust?

Maybe add this to the readiness checklist?

Reply:
I added a link to [the issue](https://github.com/NomicFoundation/edr/issues/104) in the release document